pkg/cmd/postgresdump.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2021 Ridecell, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	osExec "os/exec"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/Ridecell/ridecell-controllers/apis/db/v1beta2"
+	"github.com/Ridecell/ridectl/pkg/kubernetes"
+	"github.com/Ridecell/ridectl/pkg/utils"
+	"github.com/apoorvam/goterminal"
+	"github.com/pkg/errors"
+	"github.com/pterm/pterm"
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func init() {
+	rootCmd.AddCommand(postgresdumpCMD)
+}
+
+func getInstanceData(objName string, context string, namespace string) (string, error) {
+	var data []byte
+	var err error
+	instanceData, err := TempFS.ReadFile("template/show_postgresdump.tpl")
+	if err != nil {
+		return "", errors.Wrap(err, "error reading show_summon.tpl")
+	}
+
+	data, err = osExec.Command("kubectl", "get", "postgresdumps.db.controllers.ridecell.io", objName, "-n", namespace, "--context", context, "-o", "go-template="+string(instanceData)).Output()
+	if err != nil {
+		return "", errors.Wrap(err, err.Error())
+	}
+
+	return string(data), err
+}
+
+var postgresdumpCMD = &cobra.Command{
+	Use:   "postgresdump [flags] <microservice_name> <backup_name>",
+	Short: "Take postgres DB dump",
+	Long:  `Take postgres DB dump`,
+	Args: func(_ *cobra.Command, args []string) error {
+		if len(args) == 0 {
+			pterm.Error.Printf("Microservice name argument is required.")
+			os.Exit(1)
+		}
+		if len(args) == 1 {
+			pterm.Error.Printf("Backup name argument is required.")
+			os.Exit(1)
```

Comment:
don't exit here. if backup name is not provided, use `${microservice-name}-${epoch-time}`